### PR TITLE
ProtonMail Bridge IMAP integration

### LINKED ISSUE #429: ProtonMail Bridge IMAP integration
Using the newly released ProtonMail Bridge I am unable to add an IMAP account, with the error:

"Sorry, this folder does not exist. (IMAP)"

Log contents:

```
----------IMAP----------

* OK [CAPABILITY IMAP4rev1 STARTTLS AUTH=PLAIN AUTH=LOGIN ID APPENDLIMIT] IMAP4rev1 Service Ready
1 STARTTLS
1 OK Begin TLS negotiation now
2 LOGIN "photosphere@protonmail.ch" *********
* CAPABILITY IMAP4rev1 AUTH=PLAIN AUTH=LOGIN ID XSTOP APPENDLIMIT
2 OK [CAPABILITY IMAP4rev1 IDLE MOVE SPECIAL-USE ID XSTOP QUOTA APPENDLIMIT=26214400] LOGIN completed
3 LIST "" ""
* LIST (\Noselect) "/" "/"
3 OK LIST completed
4 ID NIL
* ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")
4 OK ID completed
ID (support-url https://protonmail.com/support name ProtonMail vendor "Proton Technologies AG")
��5 LIST "" "*"
�H�Y|� 
```
The Bridge is required to provide a decryption mechanism for ProtonMail. I've followed the instructions that might apply to a Thunderbird account, found here:

https://protonmail.com/bridge/thunderbird-mac#3e

##### Are there any related issues?

Question was originally asked in closed Issue #367, with another user experiencing the same issue. 

##### What operating system are you using?

OSX 10.11.6

##### What version of Mailspring are you using?

Version 1.0.10 (1.0.10)

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?

None

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

ProtonMail Bridge

**Feature Request?**
##### Does this feature exist in another mail client or tool you use?
Might be a feature request, considering Mailspring seems to think that there aren't any selectable folders in the account. The bridge appears to be connecting as expected, and the Thunderbird works as expected.

### PR DESCRIPTION
**Pull request by bl00mber**
resolve Foundry376/Mailspring#429 https://community.getmailspring.com/t/protonmail-bridge-imap/485
Foundry376/Mailspring#2323

### CODE REVIEW

**Comment by CodeMouse92** [comment]
WHAT!? Awesoooooooooome! Hey @bengotow, check this out. (I haven't reviewed it yet)

**Comment by bengotow** [comment]
Hey @bl00mber thanks for submitting this! It looks great - was this really the only blocker to Protonmail support? I didn't realize it was hung up on the folder hierarchy + delimiters! I looked into this back when Protonmail Bridge launched and was seeing some strange IMAP command parsing errors - at the time it seemed like their bridge was sending slightly nonstandard IMAP responses.

Will take a more detailed look at this tomorrow! 🙌

**Comment by bl00mber** [comment]
Thanks!

File: MailSync/MailUtils.cpp [hunk 1]
```diff
@@ -346,9 +347,15 @@ string MailUtils::roleForFolderViaPath(string mainPrefix, IMAPFolder * folder) {
...
     }
-
+    
            mailspringPrefix = MAILSPRING_FOLDER_PREFIX_V2 + delimiter;
     transform(mailspringPrefix.begin(), mailspringPrefix.end(), mailspringPrefix.begin(), ::tolower);
+    if (path.size() > mailspringPrefix.size() && path.substr(0, mailspringPrefix.size()) == mailspringPrefix) {
+        return path.substr(mailspringPrefix.size());
+    }
+    
+           mailspringPrefix = containerFolderPath + delimiter;
```
Comment (bengotow) [inline MailSync/MailUtils.cpp:358]: We should probably only check this scenario if `containerFolderPath` is non-empty, I think for providers that do support nested folders I think testing `/Snoozed` as well as `Mailspring/Snoozed` could cause us to detect folders at the root level that we weren't previously?

File: MailSync/Models/Account.cpp [hunk 2]
```diff
@@ -167,3 +167,7 @@ vector<string> Account::columnsForQuery() {
 void Account::bindToQuery(SQLite::Statement * query) {
     assert(false);
 }
+
+string Account::containerFolder() {
+    return _data["containerFolder"].get<string>();
```
Comment (bengotow) [inline MailSync/Models/Account.cpp]: I can't quite remember, but I think we may need to do a `_data.count("containerFolder") ? ... : ""` here, I think that this blows up if containerFolder isn't present, which might be the case for folks upgrading depending on the stuff on the JS side.

**Review by bengotow — APPROVED** [review]
I think there are a few small nits here but overall this looks awesome!  It's interesting that ProtonMail accepts the prefix "Folders/" given that it doesn't support folder hierarchies - I guess that's sort of a special prefix that it knows to strip away... I wonder if that behavior will be consistent among other providers that don't support nested folders. (I can't remember offhand, but I know there are one or two other providers that will benefit from this too!) 

Another approach to this would be to add a "use top level folders" option that makes the sync engine create `Snoozed` instead of `Mailspring/Snoozed`, but it does seem like having full control over the path prefix is a bit nicer!

**Comment by arch-btw** [comment]
Thank you so much @bl00mber ! Looks great.

**Comment by CodeMouse92** [comment]
@bl00mber Do you know if this will resolve either of these related bugs?

https://community.getmailspring.com/t/protonmail-new-emails-move-automatically-between-archive-and-inbox/210

https://community.getmailspring.com/t/protonmail-all-emails-disappear-from-inbox-and-reappear-then-disappear-again/309

**Comment by bl00mber** [comment]
@CodeMouse92 I checked these today, seems they are needed to be debugged separately so I'm converting those PRs to drafts while I'm trying to do that.

**Comment by bl00mber** [comment]
I have fixed this: https://community.getmailspring.com/t/protonmail-all-emails-disappear-from-inbox-and-reappear-then-disappear-again/309
As for the second one, wasn't able to reproduce. May be this also fixing it.

**Comment by gkaskonas** [comment]
Can this be merged?

**Comment by K-MeLeOn** [comment]
All my protonmail mail disappear like this : https://community.getmailspring.com/t/protonmail-all-emails-disappear-from-inbox-and-reappear-then-disappear-again/309/13

This correction has not been merged yet ?

**Comment by K-MeLeOn** [comment]
@bengotow Can you ping anyone who are able to review this update ?

**Comment by armaccloud** [comment]
Hi @bengotow,
Is there something I can do to move this forward? I am not a programmer, but I have a protonmail account, so could test this version if I get instructions on how to install mailspring with this fix included (Ubuntu 20.10)

**Comment by K-MeLeOn** [comment]
Hello, unfortunately, the mails disappear after more time, but disappear anyway. (1.9.1-30ef802f)

**Comment by K-MeLeOn** [comment]
@bl00mber Syncing problem persist, do you have any idea ? I'm disappointed, paying for a non functional application is very frustrating…

**Comment by bl00mber** [comment]
@K-MeLeOn stop spamming, don't you see that the PR is not merged? If you do not like the fact that the project owner do not want to review this PR for now and seemingly do not even notified of comments here - you can try to convince him to review it, email is placed in his profile.

**Comment by hallh** [comment]
I spent quite some time building and testing this today, combined with the [related PR on the Mailspring repo](https://github.com/Foundry376/Mailspring/pull/2323). I am only able to have the `Snoozed` directory created on Protonmail, but the email won't sync (see attached log below).

Also, while running the patched versions, I found that the Gmail OAuth connections become unstable as well. In this specific case, I had those set up before switching branches and replacing the patched libs, and the connection became unstable after launching the patched version. I'm unsure whether there's a direct connection to this patch or not, but it's worth mentioning.

**Log**
[mailspring-error.txt](https://github.com/Foundry376/Mailspring-Sync/files/7001520/mailspring-error.txt)

**Comment by bl00mber** [comment]
@hallh I do not have such error. Have you deleted your local database before testing? I did not but you can try to.

**Comment by hallh** [comment]
@bl00mber I haven't tried to delete the database, but I can attempt this in the weekend and see if it helps.

Not sure if this belongs here, or whether I should raise a new issue, but since PM support was recently released as of `1.9.2-6e14dad1`, I did the update and have been using it daily since.

The folders are not showing up for the PM inbox on the left-side menu, though mail is coming through. However, the contents of the PM inbox seems to appear and disappear sporadically.

I can force a sync and everything will show up, but if I switch between inboxes, the contents of the inbox may disappear again.

This only affects the PM inbox, Gmail and others are unaffected.

The logs are extensive, any suggestions for what I might be looking for to explain/fix this?

**Comment by mike-lloyd03** [comment]
@bengotow Respectfully, can we get this PR merged? Mailspring is by far the best Linux email client but the disappearing/reappearing Proton Mail issue is the thing keeping me from using it full time.

**Comment by JonathanTreffler** [comment]
Can we PLEASE get this merged?

**Comment by bl00mber** [comment]
Here is a Linux build of patched version.
@armaccloud @mike-lloyd03 @alexcanepa @jamesgecko please confirm that it works.
You need to replace `Mailspring.app/Contents/Resources/app.asar.unpacked/mailsync`
[2_Dec_mailsync.zip](https://github.com/Foundry376/Mailspring-Sync/files/7642217/mailsync.zip)

**Comment by bengotow** [comment]
Hey folks! Ahh my gosh, I thought that this had merged alongside the other one but I guess not. Let's ship this right now 👍

**Comment by mike-lloyd03** [comment]
Thanks so much!

**Comment by gkaskonas** [comment]
Finally!

**Comment by bengotow** [comment]
Hey folks! Quick update here - I merged this and made one small change, moving the "if folder is All Mail, skip" check up to the parent function. Gmail also has an "All Mail" folder, so we need to scope the if statement a bit more to avoid breaking anything I think. (5d9bbd42a77961ed3205284da7476f587cd23065) 

I merged a few other open PRs and ran a 1.9.3 build - want to give this a spin and make sure it looks good? Also going to test that the "no container folder settings" case and sync with other providers is unchanged.

If it's all good I'll publish this update tomorrow. So sorry for the hassle here guys, I had forgotten this PR was split across both repos. 

https://mailspring-builds.s3.amazonaws.com/client/ac79e444/osx/Mailspring.zip
https://mailspring-builds.s3.amazonaws.com/client/ac79e444/win-ia32/MailspringSetup.exe
https://mailspring-builds.s3.amazonaws.com/client/ac79e444/linux/mailspring-1.9.3-0.1.x86_64.rpm
https://mailspring-builds.s3.amazonaws.com/client/ac79e444/linux/mailspring-1.9.3-amd64.deb

**Comment by K-MeLeOn** [comment]
@bengotow Access denied on darwin and win32 :)

**Comment by bengotow** [comment]
Ahh thanks @K-MeLeOn - fixed them above!

**Comment by K-MeLeOn** [comment]
@bengotow top! Thanks, I'm using CanaryMail while waiting for a stable connectivity with Proton. I will try your new version :)

EDIT: for the moment I only found one issue, my protonmail boxes are on scanning everytime (see image bellow)
scanning

**Comment by ipbody** [comment]
I have the exact same problem with the snap version (lastest edge) on ubuntu 20.04

How can i get logs of this?

**Comment by armaccloud** [comment]
For me the same, I did a fresh installation of Pop!_OS 21.04 & above Mailspring deb-file, using the Protonmail Bridge.
Mailspring was using high CPU for a while syncing the different folders, now it only remains for the "All Mail" that keeps scanning.

image

**Comment by bl00mber** [comment]
See [comment](https://github.com/Foundry376/Mailspring-Sync/commit/5d9bbd42a77961ed3205284da7476f587cd23065#r60828153)
I dont think it is actually scanning, just showing status.

**Comment by armaccloud** [comment]
@bl00mber 
I am curious whether this is the case, as I get a continuous 40%-65% CPU usage from Mailspring while it doesn't seem to do anything else.

image


When I open the process, it shows the following. The _Waiting Channel_ changes and cycles between

- 0
- futex_wait_queue_me
- do_poll.constprop.0

image

ADDITION:
To test, I have removed the protonmail account from Mailspring, then added another (non-protonmail account): the CPU issue is **not** occurring.
Added back the protonmail account and the CPU issue is back.

Interesting note: The CPU **only occurs** when Mailspring is shown on-screen. Not when it is minimized.

**Comment by bl00mber** [comment]
@armaccloud [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-974634705) is a Linux x86-64 build with change from [this](https://github.com/Foundry376/Mailspring-Sync/commit/5d9bbd42a77961ed3205284da7476f587cd23065#r60828153) comment.
Currently I do not have PM subscription. Check this out, is All Mail folder still scanning on this version?

**Comment by bengotow** [comment]
Hey folks! I think that the "All Mail" folder is showing as "scanning" because we're intentionally skipping that folder during sync (since it contains a copy of the messages in all the other folders and syncing it causes them to flicker back and forth in mailspring), but we're not updating the status.

I think maybe we could hide that folder entirely? It might be that sync status < 100% is also causing the sync process to think that it needs to keep syncing as fast as it can rather than falling into an idle state. I'll take a look and see.

**Comment by armaccloud** [comment]
> @armaccloud [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-974634705) is a Linux x86-64 build with change from [this](https://github.com/Foundry376/Mailspring-Sync/commit/5d9bbd42a77961ed3205284da7476f587cd23065#r60828153) comment. Currently I do not have PM subscription. Check this out, is All Mail folder still scanning on this version?

@bl00mber Can you give me directions on how to install this? I am on Linux, the version I now have installed is the .deb file from [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-975834596).
Your .zip contains a file called mailsync. What should I do with this file?

**Comment by bengotow** [comment]
Hey folks - could you give this build a spin and see if it addresses the All Mail "syncing" issue and also the heavy CPU use? I think that this should correctly mark the folder as "sync finished", which will also let Mailsync switch to it's idle mode.

One thing I'm curious about - the All mail folder is still visible and moving mail to All Mail might do weird things, since we're not actually syncing the folder. Would be curious to know if it'd be better to just hide the folder entirely, or if this works pretty well?
 
https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/osx/Mailspring.zip
https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/linux/mailspring-1.9.3-0.1.x86_64.rpm
https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/linux/mailspring-1.9.3-amd64.deb

**Comment by ipbody** [comment]
> Hey folks - could you give this build a spin and see if it addresses the All Mail "syncing" issue and also the heavy CPU use? I think that this should correctly mark the folder as "sync finished", which will also let Mailsync switch to it's idle mode.
> 
> One thing I'm curious about - the All mail folder is still visible and moving mail to All Mail might do weird things, since we're not actually syncing the folder. Would be curious to know if it'd be better to just hide the folder entirely, or if this works pretty well?
> 
> https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/osx/Mailspring.zip https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/linux/mailspring-1.9.3-0.1.x86_64.rpm https://mailspring-builds.s3.amazonaws.com/client/9ee21cf2/linux/mailspring-1.9.3-amd64.deb

Using the .deb file i have the same problem, the All Mail folder is still scanning for ever, no progress (%) is showed

**Comment by armaccloud** [comment]
Hi @bengotow 
The same result for me as for @ipbody. I followed the steps.

- Uninstalled Mailspring
- apt autoremove
- Installed provided (last) .deb
- Verified the issues as described [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979028148) are still occurring

@bl00mber I just found the folder you were referring to in your post to replace the mailsync file. Will test.

**Comment by armaccloud** [comment]
> Here is a Linux build of patched version. @armaccloud @mike-lloyd03 @alexcanepa @jamesgecko please confirm that it works. You need to replace `Mailspring.app/Contents/Resources/app.asar.unpacked/mailsync` The size difference is because it bundles libs with it, or because some another script that I do not found is used to build prod version. [26_Nov_mailsync.zip](https://github.com/Foundry376/Mailspring-Sync/files/7604956/mailsync.zip)

I just tested the mailsync file. The same issues are occuring for me (high CPU & All Mail Scanning = visible).
What I did was

- Closed Mailspring (no process running)
- Navigate to the folder app.asar.unpacked (nautilus opened with sudo, as the folder is protected)
- Renamed the existing file to mailsync.old
- Copied the mailsync file you supplied into the folder
- Started Mailspring
- Verified the same behaviors as [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979028148)

image

**Comment by bl00mber** [comment]
@armaccloud see updated comments, I reproduced it without an account and seemingly fixed.
You can test updated version 27_Nov from the comment above.

**Comment by armaccloud** [comment]
> @armaccloud see updated comments, I reproduced it without an account and seemingly fixed.

Ah, good news & congrats!
Let me know if you need me to test anything.

**Comment by armaccloud** [comment]
> @armaccloud see updated comments, I reproduced it without an account and seemingly fixed. You can test updated version 27_Nov from the comment above.

I have tested the configuration. The scanning issue is solved, including the high cpu!

When trying to send a message, I get an authentication error on SMTP, even though I am using the provided password from the bridge (double-checked) and the settings are good.

server: 127.0.0.1
port: 1025
security: STARTTLS
allow insecure ssl: ticked
username: as provided in the bridge settings
password: as provided in the bridge settings

I see in the logs that it defaults to plain text for authentication (which is ok I guess, since it happens locally?)
When I untick _allow insecure ssl_, I get a (self-signed) certificate error

What could be the issue here? IMAP works fine (same uname & passwd)

**Error log for SMTP**

----------SMTP----------
connect 127.0.0.1 1025
220 127.0.0.1 ESMTP Service Ready
init
EHLO pop-os
250-Hello pop-os
250-PIPELINING
250-8BITMIME
250-ENHANCEDSTATUSCODES
250-CHUNKING
250-STARTTLS
250 AUTH LOGIN PLAIN
start TLS
STARTTLS
220 2.0.0 Ready to start TLS
done
init after starttls
EHLO pop-os
250-Hello pop-os
250-PIPELINING
250-8BITMIME
250-ENHANCEDSTATUSCODES
250-CHUNKING
250 AUTH PLAIN LOGIN


SASL_PATH: 

SMTP Last Response Code: 250
SMTP Last Response: Hello pop-os
PIPELINING
8BITMIME
ENHANCEDSTATUSCODES
CHUNKING
AUTH PLAIN LOGIN


mailsmtp Last Error Code: 296`

**Comment by bl00mber** [comment]
@armaccloud may be related to [this](https://github.com/Foundry376/Mailspring/issues/1996)
Is SMTP works on account from other mail providers? (In my case works)
Is SMTP works on version that Ben attached [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979511350)? (with scanning issue)
What is the log file you are copying from?

**Comment by ipbody** [comment]
> @armaccloud see updated comments, I reproduced it without an account and seemingly fixed. You can test updated version 27_Nov from the comment above.

Is just copying the mailsync file replacing the old one? when i do that, it shows me a database error that i can get rid of.
Do i have to use the stable mailspring version or the beta one from up [there](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979511350)

**Comment by armaccloud** [comment]
@ipbody 
The version .deb. you refer to is the one I used as well.
These are the steps I took to make it work.

- Closed Mailspring (no process running)
- Navigate to the folder app.asar.unpacked (nautilus opened with sudo, as the folder is protected)
- Renamed the existing file to mailsync.old
- Copied the mailsync file you supplied into the folder
- Started Mailspring

**Comment by armaccloud** [comment]
> @armaccloud may be related to [this](https://github.com/Foundry376/Mailspring/issues/1996) Is SMTP works on account from other mail providers? (In my case works) Is SMTP works on version that Ben attached [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979511350)? (with scanning issue) What is the log file you are copying from?

Hi @bl00mber 
These are the steps I took to answer your questions.

### Steps to Reproduce

- Uninstalled Mailspring completely, including config/residue
- Installed .deb from [here](https://github.com/Foundry376/Mailspring-Sync/pull/3#issuecomment-979511350) (with scanning issue)
- Added a non-protonmail account that worked earlier
- **Succes**, SMTP works, I can send emails
- Added ProtonMail account
- **Succes**, SMTP works, I can send emails from my ProtonMail account

To replace the mailsync file, took the following steps

- Closed Mailspring (no process running)
- Navigate to the folder app.asar.unpacked (nautilus opened with sudo, as the folder is protected)
- Renamed the existing file to mailsync.old
- Copied the mailsync file you supplied into the folder
- Started Mailspring
- **Error** when trying to send an email from non-ProtonMail account
- **Error** when trying to send an email from ProtonMail account

_So the issue specifically occurs after replacing the mailsync file._

To retrieve the logfile, I go to

- `Edit` - `Preferences` - `Accounts` - _highlight either non-ProtonMail or ProtonMail account_ - `Update connection settings`
- Verify the settings
- Click `Connect Account`
- Verify authentication error occurs
- Click `View Log`
- Verify logfiles

### Screenshots
image
image
image

**Comment by bl00mber** [comment]
@armaccloud thanks for explanation! Attached above 2_Dec exec build from latest commit. It is 17kb larger, maybe it helps. But I don't think it will solve the issue. It still smaller than a Ben version by 54kb... this may be something build-related, I need to investigate. For some reason SMTP works on my system.

**Comment by armaccloud** [comment]
> @armaccloud thanks for explanation! Attached above 2_Dec exec build from latest commit. It is 17kb larger, maybe it helps. But I don't think it will solve the issue. It still smaller than a Ben version by 54kb... this may be something build-related, I need to investigate. For some reason SMTP works on my system.

I have just tried out this new 2 Dec build, but the SMTP issue is still occurring.
Let me know if there are troubleshooting steps you want me to try.

**Comment by armaccloud** [comment]
Hi @bl00mber, I am wondering whether there is anything I can do to help solve this issue? It is occurring on a fresh install of both the operating system and Mailspring and the Protonmail Bridge, so I suspect I will not be the only one experiencing this. Are there any logs I can send you?

**Comment by bl00mber** [comment]
@bengotow can you build this change on your setup?

**Comment by bl00mber** [comment]
@armaccloud try to build it on your own machine.
```sh
git clone -b bl00mber-patch-1 git@github.com:bl00mber/Mailspring-Sync.git
cd <folder>
./build.sh
```
You may get some errors, try to debug them with web first.

**Comment by fuzzylogicltd** [comment]
I've just installed 1.10.0 and I'm having the "emails dissapear and reappear issue" [as descibed here](https://community.getmailspring.com/t/protonmail-all-emails-disappear-from-inbox-and-reappear-then-disappear-again/309/15). Did anyone figure out a solution for this?

**Comment by oktayacikalin** [comment]
Please, guys, try to fix this. This disappear and reappear issue really annoys me. Thanks!

**Comment by marzark** [comment]
same here

**Comment by bl00mber** [comment]
Have you tried version that I posted above, which is not working on @armaccloud system? Run that version from command line, maybe it will point to libraries that are missing? Is anyone able to reverse-engineer binary and check what library is missing in my build, or make build on his own machine, which will match the size of @bengotow binary?

**Comment by oktayacikalin** [comment]
The build.sh tries to install files into / etc.. Is it possible to test it without touching files out of my home dir? And how can I test it? Is there some quick howto?
I'm on elementary OS 6.1 (Ubuntu 20.04). Mailspring is 1.10.3 installed via apt.

**Comment by armaccloud** [comment]
> Have you tried version that I posted above, which is not working on @armaccloud system? Run that version from command line, maybe it will point to libraries that are missing? Is anyone able to reverse-engineer binary and check what library is missing in my build, or make build on his own machine, which will match the size of @bengotow binary?

Hi @bl00mber, my apologies for getting back so late. I have not yet tried building the file you sent me. I get the following error message, so maybe it is not available anymore? I have some time these weeks for troubleshooting if this works for you as well.

```
git clone -b bl00mber-patch-1 git@github.com:bl00mber/Mailspring-Sync.git
Cloning into 'Mailspring-Sync'...
The authenticity of host 'github.com (140.82.121.4)' can't be established.
ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by bl00mber** [comment]
@armaccloud you need to set up your ssh keys to be able to download from github, or you can download repo manually as archive from corresponding web page

**Comment by armaccloud** [comment]
Hi @bl00mber,
I am not very technical, could you explain the steps I have to take in order to test this?
I don't know what setting up my ssh keys means.

**Comment by lukeycold** [comment]
I've cloned it using `git clone -b bl00mber-patch-1 https://github.com/bl00mber/Mailspring-Sync.git` instead. However, I keep getting an error that says 
`Installing openssl-1.1.0f
make: *** No rule to make target 'install'.  Stop.`
These are the only two lines that appear after trying to run `./build.sh`. I'm not sure why it's happening. I'm running Fedora 36, I also tried to run the same thing under distrobox with Ubuntu to see if that might work and still nothing. I'd love to help pinpoint where the issue might be as I would really love to use this client over native protonmail.

**Comment by luckycold** [comment]
@bl00mber I've cloned it using `git clone -b bl00mber-patch-1 https://github.com/bl00mber/Mailspring-Sync.git` instead. However, I keep getting an error that says
```
Installing openssl-1.1.0f
make: *** No rule to make target 'install'.  Stop.
```
These are the only two lines that appear after trying to run `./build.sh`. I'm not sure why it's happening. I'm running Fedora 36, I also tried to run the same thing under distrobox with Ubuntu to see if that might work and still nothing. I'd love to help pinpoint where the issue might be as I would really love to use this client over native protonmail.

**Pull request merged**